src/main/java/com/kaziamyr/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.kaziamyr.onlinebookstore.service.impl;
+
+import com.kaziamyr.onlinebookstore.dto.OrderDto;
+import com.kaziamyr.onlinebookstore.dto.OrderItemDto;
+import com.kaziamyr.onlinebookstore.dto.ShippingAddressDto;
+import com.kaziamyr.onlinebookstore.exception.EntityNotFoundException;
+import com.kaziamyr.onlinebookstore.mapper.CartItemMapper;
+import com.kaziamyr.onlinebookstore.mapper.OrderItemMapper;
+import com.kaziamyr.onlinebookstore.mapper.OrderMapper;
+import com.kaziamyr.onlinebookstore.model.CartItem;
+import com.kaziamyr.onlinebookstore.model.Order;
+import com.kaziamyr.onlinebookstore.model.OrderItem;
+import com.kaziamyr.onlinebookstore.model.ShoppingCart;
+import com.kaziamyr.onlinebookstore.model.User;
+import com.kaziamyr.onlinebookstore.repository.OrderItemRepository;
+import com.kaziamyr.onlinebookstore.repository.OrderRepository;
+import com.kaziamyr.onlinebookstore.service.OrderService;
+import com.kaziamyr.onlinebookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemMapper cartItemMapper;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public OrderDto create(Map<String, String> requestBody) {
```

Comment:
This is the bad written function it has a lot of responsibilities, please introduce some methods